service/src/main/java/bio/terra/profile/service/iam/SamService.java

```diff
@@ -166,6 +163,84 @@ public List<UUID> listProfileIds(AuthenticatedUserRequest userRequest)
     }
   }
 
+  public List<SamPolicyModel> retrieveProfilePolicies(
+      AuthenticatedUserRequest userRequest, UUID resourceId) throws InterruptedException {
+    ResourcesApi resourceApi = samResourcesApi(userRequest.getToken());
+    try {
+      List<AccessPolicyResponseEntryV2> policies =
+          SamRetry.retry(
+              () ->
+                  resourceApi.listResourcePoliciesV2(
+                      SamResourceType.PROFILE.getSamResourceName(), resourceId.toString()));
+      return policies.stream()
+          .map(
+              entry ->
+                  new SamPolicyModel()
+                      .name(entry.getPolicyName())
+                      .members(entry.getPolicy().getMemberEmails()))
+          .collect(Collectors.toList());
+    } catch (ApiException e) {
+      throw SamExceptionFactory.create("Error retrieving profile policies from Sam", e);
+    }
+  }
+
+  public SamPolicyModel addProfilePolicyMember(
+      AuthenticatedUserRequest userReq, UUID resourceId, String policyName, String userEmail)
+      throws InterruptedException {
+    try {
+      return SamRetry.retry(
+          () -> addProfilePolicyMemberInner(userReq, resourceId, policyName, userEmail));
+    } catch (ApiException e) {
+      throw SamExceptionFactory.create("Error adding profile policy member in Sam", e);
+    }
+  }
+
+  private SamPolicyModel addProfilePolicyMemberInner(
+      AuthenticatedUserRequest userReq, UUID resourceId, String policyName, String userEmail)
+      throws ApiException {
+    ResourcesApi samResourceApi = samResourcesApi(userReq.getToken());
+    String samResourceName = SamResourceType.PROFILE.getSamResourceName();
+    logger.debug(
```

Comment:
Opinions wanted. Should I:

1. Make this "info" instead of "debug".
2. Not log it at all.
3. Leave as-is.

Reply:
I'd be okay with not logging it at all, the Apache proxy logs will include this information since it's all part of the URL

Reply:
I agree that capturing it elsewhere seems adequate.

A lot of http clients have logging built in too, and ultimately, there's nothing going on here that you wouldn't be able to discern from the original request.

Reply:
Will remove it.